File: jm-redis-cacheable/src/main/java/com/example/jmrediscacheable/service/service/impl/UserServiceImpl.java
package com.example.jmrediscacheable.service.service.impl;

import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl;
import com.example.jmrediscacheable.redis.annotation.RedisCacheEvict;
import com.example.jmrediscacheable.redis.annotation.RedisCacheable;
import com.example.jmrediscacheable.service.domain.User;
import com.example.jmrediscacheable.service.mapper.UserMapper;
import com.example.jmrediscacheable.service.service.UserService;
import org.springframework.stereotype.Service;

@Service
public class UserServiceImpl extends ServiceImpl<UserMapper, User>
        implements UserService {

    private final String USER_KEY = "user";

    @RedisCacheable(prefix = USER_KEY, suffix = "#id")
    @Override
    public User getUser(Long id) {
        return getById(id);
    }

    @RedisCacheEvict(prefix = USER_KEY, suffix = "#user.id")
    @Override
    public Boolean updateUser(User user) {
        return updateById(user);
    }
}

File: kg-redis-cacheable/src/main/kotlin/com/example/kgrediscacheable/serve/service/impl/UserServiceImpl.kt
package com.example.kgrediscacheable.serve.service.impl

import com.baomidou.mybatisplus.extension.service.impl.ServiceImpl
import com.example.jgrediscacheable.serve.service.UserService
import com.example.kgrediscacheable.redis.annotation.RedisCacheEvict
import com.example.kgrediscacheable.redis.annotation.RedisCacheable
import com.example.kgrediscacheable.serve.domain.User
import com.example.kgrediscacheable.serve.mapper.UserMapper
import org.springframework.stereotype.Service

/**
 * @author : songtc
 * @since : 2023/12/08 14:18
 */

@Service
class UserServiceImpl : ServiceImpl<UserMapper, User>(), UserService {

    private val USER_KEY = "user"
    @RedisCacheable(prefix = "user", suffix = "#id")
    override fun getUser(id: Long): User {
        return getById(id)
    }

    @RedisCacheEvict(prefix = "user", suffix = "#user.id")
    override fun updateUser(user: User): Boolean {
        return updateById(user)
    }
}
